src/jit/codegenxarch.cpp

```diff
@@ -1367,6 +1367,15 @@ void CodeGen::genCodeForTreeNode(GenTreePtr treeNode)
             {
                 assert(!isRegCandidate);
 
+#if defined(FEATURE_SIMD) && defined(_TARGET_X86_)
+                // Loading of TYP_SIMD12 (i.e. Vector3) variable
+                if (treeNode->TypeGet() == TYP_SIMD12)
+                {
+                    genLoadLclTypeSIMD12(treeNode);
+                    break;
+                }
+#endif // defined(FEATURE_SIMD) && defined(_TARGET_X86_)
```

Comment:
Rather than special casing TYP_SIMD12 locals in GT_LCL_VAR/STORE_LCL_VAR and spill/reload code paths, another option to consider is to lower arg accesses as GT_IND(LCL_VAR_ADDR) or GT_STOREIND(LCL_VAR_ADDR) and thereby eliminating TYP_SIMD12 local accesses.  Obviously this has the CQ downside that TYP_SIMD12 args are not enregistered and accessed from their stack location each time.  Something to consider as a short term SIMD bring up on x86.

---
In reply to: [87882442](https://github.com/dotnet/coreclr/pull/8109#discussion_r87882442) [](ancestors = 87882442,87874965,87874711)